test/e2e/e2e_test.go

```diff
@@ -252,45 +293,264 @@ var _ = Describe("e2e external gateway connectivity", func() {
+				break
+			}
+			time.Sleep(time.Second * 3)
+			framework.Logf("Retry attempt %d to get pod IP from initializing pod %s", i, dstPingPodName)
+		}
+		// Fail the test if no address is ever retrieved
+		if pingTarget == "<no value>" {
+			framework.Failf("Warning: Failed to get an IP for target pod %s, test will fail", dstPingPodName)
 		}
-		// non-HA CI mode runs a named set of nodes with a prefix of ovn-worker
-		if !haMode {
-			By(fmt.Sprintf("Creating a container on %s and verifying connectivity to the external gateway on %s", ovnWorkerNode2, ovnWorkerNode))
-			framework.ExpectNoError(
-				checkConnectivityPingToHost(f, ovnWorkerNode2, "ext-gateway-ha-ci", pingTarget, IPv4PingCommand, 30))
+		// Fail the test if the destination IP is empty
+		if pingTarget == "" {
+			framework.Failf("Warning: Failed to get an IP for target pod %s, failing test", dstPingPodName)
 		}
+		// Spin up another pod that attempts to reach the previously started pod on separate nodes
+		framework.ExpectNoError(
+			checkConnectivityPingToHost(f, ciWorkerNodeSrc, "e2e-src-ping-pod", pingTarget, ipv4PingCommand, 30))
+	})
+})
+
+// Verify pods in the namespace annotated with an external-gateway traverse the vxlan overlay and reach the
+// intended external gateway vtep. Then validate the rx counters on the vxlan interface mocking as a gateway
+var _ = Describe("e2e external gateway validation", func() {
+	var haMode bool
+	var extGatewayPodName string
+	vxlanPort := "4789"
+	svcname := "externalgw"
+	ovnNs := "ovn-kubernetes"
+	ovnWorkerNode2 := "ovn-worker2"
+	ovnControlNode := "ovn-control-plane"
+	ovnHaWorkerNode3 := "ovn-control-plane3"
+	ovnContainer := "ovnkube-node"
+	ovnNsFlag := fmt.Sprintf("--namespace=%s", ovnNs)
+	ovnContainerFlag := fmt.Sprintf("--container=%s", ovnContainer)
+	f := framework.NewDefaultFramework(svcname)
+
+	// Determine what mode the CI is running in and get relevant endpoint information for the tests
+	BeforeEach(func() {
+		labelFlag := fmt.Sprintf("name=%s", ovnContainer)
+		jsonFlag := "-o=jsonpath='{.items..metadata.name}'"
+		ipFlag := "--template={{.status.podIP}}"
+		extGW := "10.125.0.1"
+		fieldSelectorControlNodeFlag := fmt.Sprintf("--field-selector=spec.nodeName=%s", ovnControlNode)
+		fieldSelectorFlag := fmt.Sprintf("--field-selector=spec.nodeName=%s", ovnWorkerNode2)
+		fieldSelectorHaFlag := fmt.Sprintf("--field-selector=spec.nodeName=%s", ovnHaWorkerNode3)
+
+		// Attempt to retrieve the pod IP that will run the external tunnel endpoint
+		kubectlOut, err := framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorControlNodeFlag)
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnControlNode, err)
+		}
+		extGatewayPodName = strings.Trim(kubectlOut, "'")
+
+		// retrieve the ip of the node where the vxlan interface will be used as a vtep
+		vtepIP := framework.RunKubectlOrDie("get", "pods", extGatewayPodName, ipFlag, ovnNsFlag)
```

Comment:
It looks like you are creating a pod, and using that as a vxlan gw. Rather than this, can we do the real behavior for exgw and test that the ping succeeds?

1. create a docker container, with a vxlan interface and a gw ip on loopback
ip link add vxlan0 type vxlan dev eth0 id 4097 dstport 4789 remote 172.17.0.2 (ip addr of worker node with the pod on it)
ip link set vxlan0 up
ip route add 10.244.0.0/24 dev vxlan0 (10.244.0... is pod network)
ip a a 8.0.0.1/24 dev lo (gateway ip)

Then create a pod on the right worker and ping 8.0.0.1. The ping should be successful.